ci/pipeline-template.yml

```diff
@@ -0,0 +1,427 @@
+---
+resource_types:
+- name: artifactory-resource
+  type: docker-image
+  source:
+    repository: springio/artifactory-resource
+    tag:        0.0.6
+
+- name: github-status
+  type: docker-image
+  source:
+    repository: dpb587/github-status-resource
+    tag:        master
+
+- name: pull-request
+  type: docker-image
+  source:
+    repository: jtarchie/pr
+    tag:        latest
+
+- name: slack-notification
+  type: docker-image
+  source:
+    repository: nebhale/slack-notification-resource
+
+resources:
+- name: openjdk:8-jdk
+  type: docker-image
+  source:
+    repository: openjdk
+    tag:        8-jdk
+
+- name: openjdk:11-jdk
+  type: docker-image
+  source:
+    repository: openjdk
+    tag:        11-jdk
+
+- name: openjdk:12-jdk
+  type: docker-image
+  source:
+    repository: openjdk
+    tag:        12-jdk
+
+- name: spring-hateoas-github
+  type: git
+  source:
+    uri: https://github.com/spring-projects/spring-hateoas.git
+    branch: ((branch))
+
+- name: spring-hateoas-artifactory
+  type: artifactory-resource
+  source:
+    uri: https://repo.spring.io
+    username: ((artifactory-username))
+    password: ((artifactory-password))
+    build_name: spring-hateoas
+
+- name: spring-hateoas-pull-requests
+  type: pull-request
+  source:
+    access_token: ((github-access-token))
+    repo: spring-projects/spring-hateoas
+    base: ((branch))
+
+- name: spring-hateoas-status
+  type: github-status
+  source:
+    access_token: ((github-access-token))
+    repository:   spring-projects/spring-hateoas
+    branch: ((branch))
+
+- name: slack
+  type: slack-notification
+  source:
+    url: ((slack))
+    
+groups:
+- name: spring-hateoas
+  jobs:
+  - Test - JDK 8
+  - Test - JDK 8 and Spring 5.0 (snapshots)
+  - Test - JDK 8 and Spring 5.1 (snapshots)
+  - Test - JDK 11
+  - Test - JDK 11 and Spring 5.0 (snapshots)
+  - Test - JDK 11 and Spring 5.1 (snapshots)
+  - Test - JDK 12
+  - Test - JDK 12 and Spring 5.0 (snapshots)
+  - Test - JDK 12 and Spring 5.1 (snapshots)
+  - Build
+- name: pull-requests
+  jobs:
+  - spring-hateoas-pull-requests
+
+jobs:
+- name: Test - JDK 8
```

Comment:
The job configurations look awfully redundant.
Is there a way to reduce code duplication?
Like a parameterized template?

Reply:
I see the issue you cite. 

Ben says that YAML has something called "YAML references" that may support the concept needed here. I'll check it out.